src/lib/redux/inventoryDataSlice.ts

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-param-reassign */
+import { createSlice } from '@reduxjs/toolkit';
+import { InventoryRecord, InventoryUpdateRecord, ProductRecord, PurchaseRequestRecord } from '../airtable/interface';
+
+export interface SiteInventoryData {
+  siteInventory: InventoryRecord[],
+  purchaseRequests: PurchaseRequestRecord[],
+  inventoryUpdates: InventoryUpdateRecord[],
+}
+
+const EMPTY_SITE_INVENTORY_DATA : SiteInventoryData = {
+  siteInventory: [],
+  purchaseRequests: [],
+  inventoryUpdates: [],
+}
+
+interface inventoryDataSliceState {
+  products: Record<string, ProductRecord>;
+  sitesInventory: Record<string, SiteInventoryData>;
```

Comment:
It's unclear what `string` refers to here as a key. I would probably try a soft type alias to make it easier on the devs reading later.

Something like 
```
type siteId = string;
type productId = string; 

interface inventoryDataSliceState {
  products: Record<productId, ProductRecord>;
  sitesInventory: Record<siteId, SiteInventoryData>;
}
```